ebill_paynet/models/paynet_invoice_message.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import os
+from datetime import datetime
+
+# Needs Jinja 2.10
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+from odoo import fields, models
+from odoo.modules.module import get_module_root
+
+from odoo.addons.base.models.res_bank import sanitize_account_number
+
+from ..components.api import PayNetDWS
+
+import zeep  # isort:skip
+
+
+MODULE_PATH = get_module_root(os.path.dirname(__file__))
+INVOICE_TEMPLATE_2013 = "invoice-2013A.xml"
+INVOICE_TEMPLATE_2003 = "invoice-2003A.xml"
+TEMPLATE_DIR = [MODULE_PATH + "/messages"]
+
+DOCUMENT_TYPE = {"out_invoice": "EFD", "out_refund": "EGS"}
+
+
+class PaynetInvoiceMessage(models.Model):
+    _name = "paynet.invoice.message"
+    _description = "Paynet shipment send to service"
+
+    service_id = fields.Many2one(
+        comodel_name="paynet.service",
+        string="Paynet Service",
+        required=True,
+        ondelete="restrict",
+        readonly=True,
+    )
+    invoice_id = fields.Many2one(comodel_name="account.move", ondelete="restrict")
+    attachment_id = fields.Many2one("ir.attachment", "PDF")
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("sent", "Sent"),
+            ("done", "Done"),
+            ("reject", "Reject"),
+            ("error", "Error"),
+        ],
+        default="draft",
+    )
+    ic_ref = fields.Char(
+        string="IC Ref", size=14, help="Document interchange reference"
+    )
+    # Set with invoice_id.number but also with returned data from server ?
+    ref = fields.Char("Reference N°", size=35)
```

Comment:
```suggestion
    ref = fields.Char("Reference No", size=35)
```
`N°` is for french localization 

For english you will want to use `No.`

https://en.wikipedia.org/wiki/Numero_sign

Reply:
Fixed